x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextField.java

```diff
@@ -112,10 +141,12 @@ private void validate() {
                 case TEXT_EMBEDDING:
                     validateFieldPresent(DIMENSIONS_FIELD, dimensions);
                     validateFieldPresent(SIMILARITY_FIELD, similarity);
+                    // validateFieldPresent(ELEMENT_TYPE_FIELD, elementType);
```

Comment:
This check is commented out right now to keep tests passing (for now)

Reply:
The element type is not mandatory in the service settings so I think we need to handle the case where the element type is null. In such case we should just consider that the default is float.

Reply:
[Service settings' element type](https://github.com/elastic/elasticsearch/blob/1a7c46a19778ea0f8d74e29da4a56ccd6bddc1a4/server/src/main/java/org/elasticsearch/inference/ServiceSettings.java#L41) is set to a non-null value for every text embedding model, so it should be fine to enforce this check.